### PR TITLE
Render policy value with Twig

### PR DESCRIPTION
**Pull request by ingluife**
Hi, @flyve-mdm/backend-development.

### Changes description

Refactor in policy add and update. Use twig template engine for policy value. Edit showValueInput methods of classes corresponding to each value type.

### Print screen
Add policy - Deploy file:
image

Add policy - Deploy Application
image

Update policy - Deploy file
image

### References

Related #308

### CODE REVIEW

**Comment by ingluife** [comment]
Hi, @flyve-mdm/backend-development.

This is ready for be Merged.

**Comment by btry** [comment]
Hi @ingluife 

We cannot add a grid system in the project for the reasons I gave above. We should use a table, sad but true :(

**Comment by ingluife** [comment]
Hi, @flyve-mdm/backend-development.

I changed the way of render the policy value to maquetation with table.

This is the print screem of the result in the view:

- Deploy application:
image

- Deploy file:
image

- Modal edit Deploy file:
image

- Deploy file in Mobile:
In this part, there are a scroll to right for see the other fields.
image

- Modal edit Deploy file in Mobile:
In this part, there are a scroll to right for see the other fields.
image

**Comment by ingluife** [comment]
This is ready for merge.

File: inc/policybase.class.php [hunk 1]
```diff
@@ -240,9 +240,12 @@ public function post_unapply($value, $itemtype, $itemId, PluginFlyvemdmNotifiabl
     * @return string
     */
    public function showValueInput($value = '', $itemType = '', $itemId = 0) {
-      $html = '<input name="value" value="' . $value . '" >';
 
-      return $html;
+      $data['typeTmpl'] = 'PluginFlyvemdmPolicyBase';
```
Comment (btry) [inline inc/policybase.class.php]: It is better to use PluginFlyvemdmPolicyBase::class

File: inc/policyboolean.class.php [hunk 2]
```diff
@@ -89,7 +89,13 @@ public function getMqttMessage($value, $itemtype, $itemId) {
     * @return int|string
     */
    public function showValueInput($value = '', $itemType = '', $itemId = 0) {
-      return Dropdown::showYesNo('value', $value, -1, ['display' => false]);
+      $data['typeTmpl'] = 'PluginFlyvemdmPolicyBoolean';
```
Comment (btry) [inline inc/policyboolean.class.php]: Same here

File: inc/policybase.class.php [hunk 3]
```diff
@@ -240,9 +240,12 @@ public function post_unapply($value, $itemtype, $itemId, PluginFlyvemdmNotifiabl
...
+      $data['itemtype'] = $itemType;
+      $data['value'] = $value;
+      $twig = plugin_flyvemdm_getTemplateEngine();
+      return $twig->render('tmpl_policy_value.html.twig', ['data' => $data]);
```
Comment (btry) [inline inc/policybase.class.php]: The extension .twig telle the file is a template. It is useless to prefix with tmpl_

File: inc/policyboolean.class.php [hunk 4]
```diff
@@ -89,7 +89,13 @@ public function getMqttMessage($value, $itemtype, $itemId) {
...
+      $data['itemtype'] = $itemType;
+      $data['dropdown'] = [
+          Dropdown::showYesNo('value', $value, -1, ['display' => false])
+      ];
+      $twig = plugin_flyvemdm_getTemplateEngine();
+      return $twig->render('tmpl_policy_value.html.twig', ['data' => $data]);
```
Comment (btry) [inline inc/policyboolean.class.php]: Same here

File: inc/policydeployapplication.class.php [hunk 5]
```diff
@@ -235,26 +235,19 @@ public function showValueInput($value = '', $itemType = '', $itemId = 0) {
...
+                  'displaywith'  => ['alias'],
+                  'name'         => 'items_id',
+                  'value'        => $itemId,
+               ]),
+            Dropdown::showYesNo('value[remove_on_delete]', $removeOnDelete, -1, ['display' => false])
       ];
-
       $twig = plugin_flyvemdm_getTemplateEngine();
-      return $twig->render('policy_deploy_app_form.html.twig', $data);
+      return $twig->render('tmpl_policy_value.html.twig', ['data' => $data]);
```
Comment (btry) [inline inc/policydeployapplication.class.php]: Same here

File: inc/policydeployfile.class.php [hunk 6]
```diff
@@ -279,37 +279,26 @@ public function showValueInput($value = '', $itemType = '', $itemId = 0) {
...
+            'display'   => false,
+            'name'      => 'destination_base',
+            'value'     => $path->getID(),
+            ]),
+            Dropdown::showYesNo('value[remove_on_delete]', $removeOnDelete, -1, ['display' => false])
       ];
-
       $twig = plugin_flyvemdm_getTemplateEngine();
-      return $twig->render('policy_deploy_file_form.html.twig', $data);
+      return $twig->render('tmpl_policy_value.html.twig', ['data' => $data]);
```
Comment (btry) [inline inc/policydeployfile.class.php]: Same here

File: inc/policydropdown.class.php [hunk 7]
```diff
@@ -102,7 +102,13 @@ public function translateData() {
...
     */
    public function showValueInput($value = '', $itemType = '', $itemId = 0) {
-      return Dropdown::showFromArray('value', $this->valueList, ['display' => false, 'value' => $value]);
+      $data['typeTmpl'] = 'PluginFlyvemdmPolicyDropdown';
+      $data['itemtype'] = $itemType;
+      $data['dropdown'] = [
+          Dropdown::showFromArray('value', $this->valueList, ['display' => false, 'value' => $value])
+      ];
+      $twig = plugin_flyvemdm_getTemplateEngine();
+      return $twig->render('tmpl_policy_value.html.twig', ['data' => $data]);
```
Comment (btry) [inline inc/policydropdown.class.php]: Same here

File: inc/policydropdown.class.php [hunk 8]
```diff
@@ -102,7 +102,13 @@ public function translateData() {
     * @return int|string
     */
    public function showValueInput($value = '', $itemType = '', $itemId = 0) {
-      return Dropdown::showFromArray('value', $this->valueList, ['display' => false, 'value' => $value]);
+      $data['typeTmpl'] = 'PluginFlyvemdmPolicyDropdown';
```
Comment (btry) [inline inc/policydropdown.class.php]: Same here

File: tests/suite-unit/PluginFlyvemdmPolicyBase.php [hunk 9]
```diff
@@ -123,7 +123,10 @@ public function testPre_unapply() {
     */
    public function testShowValueInput() {
       list($policy) = $this->createNewPolicyInstance();
-      $this->string($policy->showValueInput())->isEqualTo('<input name="value" value="" >');
+      $data['typeTmpl'] = 'PluginFlyvemdmPolicyBase';
```
Comment (btry) [inline tests/suite-unit/PluginFlyvemdmPolicyBase.php]: Same here

**Comment by ingluife** [comment]
Hi, @btry.
Ready the minor changes requested.

**Comment by btry** [comment]
please, take care of my ears :)

**Pull request merged**